### PR TITLE
Using sendgrind API v3

### PR DESCRIPTION
**Pull request by Nyholm**
The version of sendgrid we are using is old. This will make sure we use the latest version.  

This will fix #64

### CODE REVIEW

File: lib/Mailer/SendGrid.php [hunk 1]
```diff
@@ -157,19 +144,48 @@ protected function format(MessageInterface $message)
...
+
+        return $processedAttachments;
+    }
+
+    /**
+     * @param IdentityInterface[]|string $recipients
+     *
+     * @return array
+     */
+    public function formatRecipients($recipients)
```
Comment (stof) [inline lib/Mailer/SendGrid.php]: no reason to have it public

File: lib/Mailer/SendGrid.php [hunk 2]
```diff
@@ -25,128 +26,114 @@ class SendGrid extends Mailer
...
+        if ($message instanceof TaggableInterface) {
+            $parameters['categories'] = (array) $message->getTag();
         }
 
-        if ($smtpApi) {
-            $parameters['x-smtpapi'] = json_encode(array_filter($smtpApi));
+        if ($message instanceof MetadataAwareInterface) {
+            $metadata = array_filter($message->getMetadata());
+            if (!empty($metadata)) {
+                $parameters = array_merge($parameters, $metadata);
```
Comment (stof) [inline lib/Mailer/SendGrid.php]: This looks wrong to me. AFAICT, custom metadata should go in the `custom_args` parameter, not on top-level

**Comment by stof** [comment]
This must clearly be documented as a BC break in the readme, as this changes the format of the server token

**Comment by Nyholm** [comment]
Thank you for the reviews

**Pull request merged**